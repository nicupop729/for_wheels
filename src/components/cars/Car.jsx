import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const Car = ({ car }) => {
  const {
    model,
    price,
  } = car;
  const ImgUrl = car.img_url;
  return (
    <div className="flex flex-col p-8 inline-block shadow-lg mb-4 mx-3">
      <img src={ImgUrl} alt={`This is a${{ model }}`} />
      <span>{model}</span>
      <span>
        Price:
        {price}
        {' '}
        USD
      </span>
      <Link to="/car" state={car} className="border-solid border-2 border-dark p-6 bg-green-200">SHOW MORE</Link>
    </div>
  );
};

Car.defaultProps = {
  car: {
    model: '',
    price: '',
    img_url: '',
  },
};

Car.propTypes = {
  car: PropTypes.shape({
    model: PropTypes.string,
    price: PropTypes.number,
    img_url: PropTypes.string,
  }),
};

export default Car;
