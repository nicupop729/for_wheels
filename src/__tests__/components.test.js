import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import React from 'react';
import Car from '../components/cars/Car';

describe('Tests for React components rendering', () => {
  const car = {
    id: 11,
    img_url: 'https://elitetraveler.com/wp-content/uploads/2019/07/Screen-Shot-2019-07-05-at-10.48.37-768x486.png',
    model: 'Aston Martin Valkyrie',
    description: 'Valkyrie, its newest flagship, is a hybrid hypercar powered by an internal combustion engine. A bit of a head-scratcher, the battery pack services the vehicleâ€™s electronics and integrates all relevant sub-systems while the V12 engine does the heavy lifting.',
    price: 0.0,
    created_at: '2022-04-13T18:34:43.595Z',
    updated_at: '2022-04-13T18:34:43.595Z',
  };

  it('Tests Car component', () => {
    const tree = render(
      <BrowserRouter>
        <Car car={car} />
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
