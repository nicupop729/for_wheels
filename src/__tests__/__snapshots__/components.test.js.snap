// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for React components rendering Tests Car component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="mb-5"
        href="/car"
      >
        <div
          class="flex flex-col py-8 px-4 inline-block mx-3 min-h-full shadow-md hover:shadow-xl transition duration-700"
        >
          <div
            class="relative h-0 pb-3/7 pt-1/4 flex-1 xl:mb-8"
          >
            <img
              alt="This is a[object Object]"
              class="absolute inset-0 w-full h-full rounded-lg object-cover z-0"
              src="https://elitetraveler.com/wp-content/uploads/2019/07/Screen-Shot-2019-07-05-at-10.48.37-768x486.png"
            />
          </div>
          <span
            class="text-2xl mb-5 font-bold"
          >
            Aston Martin Valkyrie
          </span>
          <span
            class="text-xs text-gray-400"
          >
            Valkyrie, its newest flagship, is a hybrid hypercar powered by an internal combustion engine. A bit of a head-scratcher, the battery pack services the vehicle’s electronics and integrates all relevant sub-systems while the V12 engine does the heavy lifting.
          </span>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="mb-5"
      href="/car"
    >
      <div
        class="flex flex-col py-8 px-4 inline-block mx-3 min-h-full shadow-md hover:shadow-xl transition duration-700"
      >
        <div
          class="relative h-0 pb-3/7 pt-1/4 flex-1 xl:mb-8"
        >
          <img
            alt="This is a[object Object]"
            class="absolute inset-0 w-full h-full rounded-lg object-cover z-0"
            src="https://elitetraveler.com/wp-content/uploads/2019/07/Screen-Shot-2019-07-05-at-10.48.37-768x486.png"
          />
        </div>
        <span
          class="text-2xl mb-5 font-bold"
        >
          Aston Martin Valkyrie
        </span>
        <span
          class="text-xs text-gray-400"
        >
          Valkyrie, its newest flagship, is a hybrid hypercar powered by an internal combustion engine. A bit of a head-scratcher, the battery pack services the vehicle’s electronics and integrates all relevant sub-systems while the V12 engine does the heavy lifting.
        </span>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
